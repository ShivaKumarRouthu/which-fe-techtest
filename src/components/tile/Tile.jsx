import React from 'react'

import './tile.css';

const Tile = ({ num, onTileClicked, isSelected }) => {

  return (
    <div className='tile-container' >
      <button className={`btn ${isSelected?'selected': ''}`} data-testid={`tile-item-${num}`} onClick={() => onTileClicked(num)} >
        {num}
      </button>
    </div>
  )
}

export default Tile