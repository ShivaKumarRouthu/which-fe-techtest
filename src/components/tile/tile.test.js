import '@testing-library/jest-dom/extend-expect'

import {
  fireEvent,
  getByTestId,
  render,
  screen
} from '@testing-library/react'

import React from 'react'

import Tile from './Tile'


test('should render component', () => {
  const { asFragment } = render(<Tile num={1}/>)
  expect(asFragment()).toMatchSnapshot()
})


test('should render a tile ', () => {
  render(<Tile num={1} />)
  const tile = screen.getByTestId('tile-item-1')
  expect(tile).toBeInTheDocument()
})

test('should render a tile ', () => {
  render(<Tile num={2} />)
  const tile = screen.getByTestId('tile-item-2')
  expect(tile).toBeInTheDocument()
})

test('should render the tile item correctly with given props', () => {
  render(<Tile num={2} />)
  const linkElement = screen.getByTestId('tile-item-2');
  expect(linkElement).toHaveTextContent('2');
})

test('should render the tile item with class name selected', () => {
  render(<Tile num={2} isSelected={true} />)
  const linkElement = screen.getByTestId('tile-item-2');
  expect(linkElement).toHaveClass('selected');
})