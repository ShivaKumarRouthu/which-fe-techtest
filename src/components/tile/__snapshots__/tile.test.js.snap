// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<DocumentFragment>
  <div
    class="tile-container"
  >
    <button
      class="btn "
      data-testid="tile-item-1"
    >
      1
    </button>
  </div>
</DocumentFragment>
`;
